jdk-http-client/src/main/scala/org/http4s/client/jdkhttpclient/JdkHttpClient.scala

```diff
@@ -61,51 +120,12 @@ object JdkHttpClient {
       "warning"
     ).map(CaseInsensitiveString(_))
 
-  def convertRequest[F[_]: ConcurrentEffect](req: Request[F]): HttpRequest = {
-    val rb = HttpRequest.newBuilder
-      .method(
-        req.method.name, {
-          val publisher = FlowAdapters.toFlowPublisher(
-            StreamUnicastPublisher(req.body.chunks.map(_.toByteBuffer)))
-          if (req.isChunked)
-            BodyPublishers.fromPublisher(publisher)
-          else
-            req.contentLength.fold(BodyPublishers.noBody)(
-              BodyPublishers.fromPublisher(publisher, _))
-        }
-      )
-      .uri(URI.create(req.uri.renderString))
-      .version(req.httpVersion match {
-        case HttpVersion.`HTTP/1.1` => HttpClient.Version.HTTP_1_1
-        case HttpVersion.`HTTP/2.0` => HttpClient.Version.HTTP_2
-        case _ => HttpClient.Version.HTTP_1_1
-      })
-    val headers = req.headers.iterator
-    // hacky workaround (see e.g. https://stackoverflow.com/questions/53979173)
-      .filterNot(h => restrictedHeaders.contains(h.name))
-      .flatMap(h => Iterator(h.name.value, h.value))
-      .toArray
-    (if (headers.isEmpty) rb else rb.headers(headers: _*)).build
-  }
-
-  def convertResponse[F[_]: ConcurrentEffect](
-      res: HttpResponse[Flow.Publisher[util.List[ByteBuffer]]]): F[Response[F]] =
-    ConcurrentEffect[F].fromEither(Status.fromInt(res.statusCode)).map { status =>
-      Response(
-        status = status,
-        headers = Headers(res.headers.map.asScala.flatMap {
-          case (k, vs) => vs.asScala.map(Header(k, _))
-        }.toList),
-        httpVersion = res.version match {
-          case HttpClient.Version.HTTP_1_1 => HttpVersion.`HTTP/1.1`
-          case HttpClient.Version.HTTP_2 => HttpVersion.`HTTP/2.0`
-        },
-        body = FlowAdapters
-          .toPublisher(res.body)
-          .toStream[F]
-          .flatMap(bs =>
-            Stream.fromIterator(bs.asScala.map(Chunk.byteBuffer).iterator).flatMap(Stream.chunk))
-      )
-    }
+  private def convertHttpVersionFromHttp4s[F[_]](version: HttpVersion)(
+      implicit F: ApplicativeError[F, Throwable]): F[HttpClient.Version] =
+    F.fromEither(version match {
```

Comment:
You could do `F.pure` for the `Right`s and `F.raiseError` for the `Left` and skip the `fromEither`.